src/angular-app/languageforge/lexicon/settings/configuration/configuration-fields.component.html

```diff
@@ -120,42 +114,12 @@
                 </th>
                 <td></td>
                 <td></td>
-                <td class="text-middle align-middle">
-                    <button type="button" class="btn btn-std text-middle" data-ng-click="$ctrl.resetInputSystemsForRole($ctrl.unifiedViewModel.inputSystems.settings, $ctrl.unifiedViewModel.inputSystems.hasCustomInputSystemsOverride, $ctrl.unifiedViewModel.inputSystems.selectAllColumns, $ctrl.fccConfigDirty, 'observer')"
-                            data-ng-show="$ctrl.unifiedViewModel.inputSystems.hasCustomInputSystemsOverride.observer"
-                            id="reset-input-system-observer-btn" title="Reset input systems for observer">
-                        Reset
-                    </button>
-                </td>
-                <td>
-                    <button type="button" class="btn btn-std" data-ng-click="$ctrl.resetInputSystemsForRole($ctrl.unifiedViewModel.inputSystems.settings, $ctrl.unifiedViewModel.inputSystems.hasCustomInputSystemsOverride, $ctrl.unifiedViewModel.inputSystems.selectAllColumns, $ctrl.fccConfigDirty, 'commenter')"
-                            data-ng-show="$ctrl.unifiedViewModel.inputSystems.hasCustomInputSystemsOverride.commenter"
-                            id="reset-input-system-commenter-btn" title="Reset input systems for commenter">
-                        Reset
-                    </button>
-                </td>
-                <td>
-                    <button type="button" class="btn btn-std" data-ng-click="$ctrl.resetInputSystemsForRole($ctrl.unifiedViewModel.inputSystems.settings, $ctrl.unifiedViewModel.inputSystems.hasCustomInputSystemsOverride, $ctrl.unifiedViewModel.inputSystems.selectAllColumns, $ctrl.fccConfigDirty, 'contributor')"
-                            data-ng-show="$ctrl.unifiedViewModel.inputSystems.hasCustomInputSystemsOverride.contributor"
-                            id="reset-input-system-contributor-btn" title="Reset input systems for contributor">
-                        Reset
-                    </button>
-                </td>
-                <td>
-                    <button type="button" class="btn btn-std" data-ng-click="$ctrl.resetInputSystemsForRole($ctrl.unifiedViewModel.inputSystems.settings, $ctrl.unifiedViewModel.inputSystems.hasCustomInputSystemsOverride, $ctrl.unifiedViewModel.inputSystems.selectAllColumns, $ctrl.fccConfigDirty, 'manager')"
-                            data-ng-show="$ctrl.unifiedViewModel.inputSystems.hasCustomInputSystemsOverride.manager"
-                            id="reset-input-system-manager-btn" title="Reset input systems for manager">
-                        Reset
-                    </button>
-                </td>
+                <td></td>
+                <td></td>
+                <td></td>
+                <td></td>
```

Comment:
I'm thinking you could get away with a `<td colspan="6"></td>` here as an equivalent expression of table structure.

Reply:
Done, and also in all the other places where this exists. I thought about using `colspan="0"` but it seems that feature was *removed* from HTML 5, sadly, due to browsers other than Firefox not implementing it. Apparently `rowspan="0"` is still valid (this cell should extend downwards across all rows), but `colspan="0"` is not allowed as a way of saying "this cell should extend rightwards across all columns). Sigh; that means your HTML has to know how many columns it contains, and it's easy to get it wrong when modifying a table.

There's another possible improvement: everywhere where I just made `colspan="6"`, we also have `<td data-ng-repeat="group in $ctrl.unifiedViewModel.exampleFields.selectAllColumns.groups"></td>`. We could do `<td colspan="$ctrl.unifiedViewModel.exampleFields.selectAllColumns.groups.length">" here. There'd be a problem if there are no groups so the length was 0, but everywhere where this appears, there's a final `<td></td>` so we could use `length + 1` and colspan would always be at least 1. However, I'm thinking this would be a bit too clever for our own good, as later someone might remove the final column of the table and delete the `+1` and then we'd have `colspan="0"` which would cause issues in Chrome. So I'm going to leave well enough alone regarding the `ng-repeat`.

Reply:
great thanks.  Thanks also for the detailed explanation.